### PR TITLE
Add primitive arrays to Arb

### LINKED ISSUE #2300: Support Arb primitive arrays
Currently only `Arb.byteArray` is supported.
I suggest that we add support for other primitive arrays as well.

### PR DESCRIPTION
**Pull request by BartvHelvert**
Closes #2300.
Depends on #2299.

### CODE REVIEW

**Comment by sksamuel** [comment]
I think will need to be rebased.

File: kotest-property/src/commonMain/kotlin/io/kotest/property/arbitrary/longs.kt [hunk 1]
```diff
@@ -34,13 +47,20 @@ fun Arb.Companion.positiveLong(max: Long = Long.MAX_VALUE): Arb<Long> = long(1L,
...
 
-class LongShrinker(private val range: LongRange) : Shrinker<Long> {
-   override fun shrink(value: Long): List<Long> = when (value) {
-      0L -> emptyList()
-      1L, -1L -> listOf(0L).filter { it in range }
+/**
+ * Returns an [Arb] that produces [LongArray]s where [generateArrayLength] produces the length of the arrays and
+ * [generateContents] produces the content of the arrays.
+ */
+fun Arb.Companion.longArray(generateArrayLength: Gen<Int>, generateContents: Arb<Long>): Arb<LongArray> =
```
Comment (sksamuel) [inline kotest-property/src/commonMain/kotlin/io/kotest/property/arbitrary/longs.kt:54]: We tend to use the name somethingArb or somethingGen when composing arbs/gens.
So perhaps lengthGen and contentArb or something along those lines ?

Comment (BartvHelvert) [reply kotest-property/src/commonMain/kotlin/io/kotest/property/arbitrary/longs.kt:54]: I am not sure how useful it is to repeat the type in the parameter name? I think `length`/`content` would be clear enough?

Comment (sksamuel) [reply kotest-property/src/commonMain/kotlin/io/kotest/property/arbitrary/longs.kt:54]: Ok agreed.

**Pull request merged**